data-processing/scripts/dataset_validator.py

```diff
@@ -0,0 +1,259 @@
+
+    @field_validator("created")
+    @classmethod
+    def check_created_format(cls, v: str):
+        if not v:
+            raise ValueError("Timestamp 'created' is required.")
+        try:
+            start, end = v.split(", ")
+            start_date = datetime.strptime(start, "%Y-%m-%dT%H:%M:%S.%fZ")
+            end_date = datetime.strptime(end, "%Y-%m-%dT%H:%M:%S.%fZ")
+            if start_date > end_date:
+                raise ValueError(
+                    "Timestamp 'created' should be in the format 'YYYY-MM-DDTHH:MM:SS.TIMEZONE, YYYY-MM-DDTHH:MM:SS.TIMEZONE'.",
+                )
+        except ValueError:
+            raise ValueError(
+                "Timestamp 'created' should be in the format 'YYYY-MM-DDTHH:MM:SS.TIMEZONE, YYYY-MM-DDTHH:MM:SS.TIMEZONE'.",
+            )
+
+
+def check_first_entry(document_file: Path):
+    """
+    Get first entry in jsonl.gz file.
+    """
+    dataset_name = document_file.parent.parent.name
+    with gzip.open(document_file, "rb") as f:
+        line = f.readline().decode("utf-8")
+
+    json_entry = json.loads(line)
+    doc = Document(**json_entry)
+    assert (
+        doc.source == dataset_name
+    ), f"Source should be {dataset_name}, but is {doc.source}"
+
+
+def create_parser() -> argparse.ArgumentParser:
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument(
+        "--dataset_folder",
+        type=str,
+        help="Path to the dataset folder. On UCloud this is danish-foundation-models/dfm-data/pre-training",
+    )
+    parser.add_argument(
+        "--datasheets_folder",
+        type=str,
+        help="Path to the datasheets folder. On GitHub this is 'danish-foundation-models/docs/datasheets'",
+    )
+    return parser
+
+
+def convert_to_paths(dataset_folder: str, datasheets_folder: str) -> tuple[Path, Path]:
+    _dataset_folder = Path(dataset_folder)
+    _datasheets_folder = Path(datasheets_folder)
+
+    if not _dataset_folder.exists():
+        raise FileNotFoundError(
+            f"Dataset folder {_dataset_folder.resolve()} does not exist.",
+        )
+    if not _datasheets_folder.exists():
+        raise FileNotFoundError(
+            f"Datasheets folder {_datasheets_folder.resolve()} does not exist.",
+        )
+
+    return _dataset_folder, _datasheets_folder
+
+
+def check_datasheet(dataset_path: Path, datasheets_path: Path) -> str:
+    datasheet_path = datasheets_path / dataset_path.name
+    msg = ""
+
+    if not datasheet_path.exists():
+        msg += f"Datasheet {datasheet_path.name} does not exist.\n"
+
+    # extract frontmatter from datasheet
+    try:
+        with datasheet_path.open("r") as f:
+            markdown = f.read()
+            frontmatter = yaml.safe_load(markdown.split("---")[1])
+    except Exception as e:
+        msg += f"Error reading datasheet {datasheet_path.name}: {e}\n"
+        return msg
+
+    # check if datasheet contains required fields
+    required_fields = ["license", "languages"]
```

Comment:
I believe that it is `language` instead in [huggingface](https://github.com/huggingface/hub-docs/blob/main/datasetcard.md?plain=1) :)

Reply:
Fixed